### ISSUE TITLE
Square size test: blob transactions missing

### ISSUE THREAD

**Issue by ivan-gavran** [celestiaorg/celestia-app#2034]
### Summary of Bug

When running `square_size_test`, I noticed that blocks that are created are always of a very small size (mostly 1, rarely 4). Upon digging deeper, I understood there are never any `blobTx` transactions reaching `PrepareProposal`.

I don't fully understand why that is so, but there seems to be a problem in [this check](https://github.com/celestiaorg/celestia-core/blob/f5f063c9aab490a9abd9453f147ac6a285da186a/types/tx.go#L255-L257) inside `UnmarshalBlobTx`.

### Version

ce0be9516e6e7292cf41f7a073b55fab8346ca3f

### Steps to Reproduce

In the [fork of the repo](https://github.com/ivan-gavran/celestia-app/tree/ivan/square_size_test2), I added logs showing that blobs are missing. 
Run with `go test -v -timeout 300s -run TestSquareSizeIntegrationTest -testify.m TestSquareSizeUpperBound_Flaky`

### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by evan-forbes** [comment]
can we close this issue? @cmwaters do you remember if we ever found the reason for this? the test appears to be checking that the actual square size is being hit

**Comment by evan-forbes** [comment]
closing as this issue does not appear to be present. please reopen if need be

**Comment by ivan-gavran** [comment]
@evan-forbes , sorry for joining the discussion late. Could you clarify how did you check that the actual square size is being hit? In the [fork](https://github.com/ivan-gavran/celestia-app/tree/ivan/square_size_test2) I mentioned, upon running the test, I get output like this one:
```
totalSize of the block
	normal: 257
	blob: 0
proposal block 31 created with 1 txs and 257 bytes
```

**Comment by evan-forbes** [comment]
@ivan-gavran I see! thanks for bringing this back up! I'm not sure if I glanced over the greater than there or assumed it was already fixed on main relative to the fork you had on main. Changed to a require.Equal in #2446, along with decreasing the pfb size so that they don't get rejected. I think we accidently changed the test in #1768 

We're not deciphering between the errors returned by txsim there because it always returns an error, which is a whole other issue.
